### ISSUE TITLE
Urgent: When I hide last line auf fieldgroup, then data is saved anyway

### ISSUE THREAD

**Issue by dasistwas** [dasistwas/moodle-mod_datalynx#26]
If I remove the last line the line is saved anyway and the content is displayed in the view in view-mode.
Also a problem:
When I have 2 lines saved and the default number of lines displayed = 1 then only one line is shown, but the 2 lines should be shown in edit mode.

**Comment by dasistwas** [comment]
"Hide" is not a correct wording. Delete is correct. Data should not be hidden but also deleted upon saving the entry when a line is deleted.

**Comment by michaelpollak** [comment]
We don't save not visible lines but in case there is something in a line from a prior edit it is not removed by the update. It was the intended behaviour, when I hide a line I don't want to remove the content I have added last time I edited.
I will work on the remove line functionality first and use the same logic here in case we want it to do sth. else.

**Comment by dasistwas** [comment]
Sorry, I have to clarify this:
- We do not hide, we delete: Thus hiding a line deletes the content (via JS and upon save)
- Prior edits have to be deleted as well

Thank you for changing the functionality.

**Comment by dasistwas** [comment]
Hidden lines with content should not exist. So it would be great to have delete instead of hide.

**Comment by michaelpollak** [comment]
(1) we keep "show one more line" and get rid of "hide last line". 
(2) Removing is done on a line by line basis somehow, see #24 .
(3) We always show as many lines as there is content to show.

**Comment by dasistwas** [comment]
I assume this is work in progress. Observations of last test:

- Content of textbox is deleted when clicking on "Remove this line" but uploaded picture stays in place, but should also be deleted
- The line I removed is still visible, but should not be. Technically content has to be deleted and the line "hidden" or altered in a way users do not see that anymore.
- The text "Hide this line" is unclear. If I am a user, I do not see a "line". So just naming it "Delete" or "Remove would be better.

**Comment by dasistwas** [comment]
This is quite tricky. To solve we could follow one of those strategies:

1) Delegate events to the respective buttons (Click on file, Click on delete, Confirm delte)

2) Use M.form_filemanager in order to delete the files:

The magic happens there:
https://github.com/moodle/moodle/blob/master/lib/form/filemanager.js#L837
callback arguments are applied. So we have to take a shortcut to this callback skipping the dialog. So we should call:
https://github.com/moodle/moodle/blob/master/lib/form/filemanager.js#L837
`dialog_options.callback = function(params)`

But I am not sure how that is done with YUI.

**Comment by michaelpollak** [comment]
Thank you for your feedback David, 
yes removing of lines is a work in progress and currently only works with text and simple fields. I have split this issue and work on this feature on #24 see commit there.
The last line will be hidden when we remove one line, for testing purposes and to implement the complex fields I kept it visible for now. 
"Hide this line" was removed with one of the last commits, feature is obsolete.

Thank you for sharing your Know-How! Filemanager uses a hidden field to pass the files key, to me the simplest way is to remove this hidden key that is passed in $data and reset the filemanager with javascript. So for files and images it's a two step process. 
Similar problem with timestamps, there multiple fields need to be reset.
